### PR TITLE
[CD] Add Listening workflows for website publishing.

### PR DESCRIPTION
**Pull request by ProjectSynchro**
This enables automatic publishing of the website, as well as the lua documentation.

Other PRs will require a personal access token, passed in as the secret name WEBSITE_ACCESS_TOKEN.
This should have workflow permissions to trigger workflows.

This should be merged before both https://github.com/naev/naev/pull/1335 and https://github.com/naev/naev-website/pull/5

### CODE REVIEW

File: .github/workflows/website_publish.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      - name: Checkout Naev Repository
+        uses: actions/checkout@v2
+        with:
+          path: prod
+
+      - name: Clone Website
+        if: github.event.action == 'website'
+        uses: actions/checkout@v2
+        with:
+          repository: 'ProjectSynchro/naev-website'
```
Comment (bobbens) [inline .github/workflows/website_publish.yml]: Shouldn't this be `naev/naev-website`?

File: .github/workflows/website_publish.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          path: staging
+
+      - name: Clone API Docs
+        if: github.event.action == 'api'
+        uses: actions/checkout@v2
+        with:
+          repository: 'ProjectSynchro/naev'
```
Comment (bobbens) [inline .github/workflows/website_publish.yml]: `naev/naev`?

**Review by bobbens — COMMENTED** [review]
If I understand correctly, this listens on other repos and when there are changes tries to update itself right?

**Comment by bobbens** [comment]
Speaking of which, how to generate the repo key? This is not a personal token but a deploy key or webhook?

File: .github/workflows/website_publish.yml (hunk 1 above)
Comment (ProjectSynchro) [reply .github/workflows/website_publish.yml]: This should be, good catch

File: .github/workflows/website_publish.yml (hunk 2 above)
Comment (ProjectSynchro) [reply .github/workflows/website_publish.yml]: Same thing here.

**Comment by ProjectSynchro** [comment]
> If I understand correctly, this listens on other repos and when there are changes tries to update itself right?

This is exactly right, the other repositories send a trigger via the REST API to this repository, which will run certain steps depending on what the trigger name is. (Think that manual-nightly script thingy)



> Speaking of which, how to generate the repo key? This is not a personal token but a deploy key or webhook?

I think I misworded my initial message in this thread, I meant to say: you will need to create and upload a personal access token as a secret, as you'd be sending a trigger for this workflow remotely. So long as that token has 'workflow' permissions it should work. The jobs on the other repositories would trigger this as 'you' (or another service account would do as well.. so long as it has access) and then this workflow would take care of committing changes and pushing them.

**Comment by bobbens** [comment]
Have to figure out the token thing still, but looks good, thanks!

**Pull request merged**